src/resolvers/section.ts

```diff
@@ -0,0 +1,193 @@
+
+                    // Create a Set of existing tag names
+                    const existingTagIds = new Set(existingTags.map(tag => tag.name));
+
+                    // Filter out the tags that already exist in the table.
+                    const tagsToAdd = tags.filter(tag => !existingTagIds.has(tag.name));
+
+                    // Add tags to sectionTags table that did not already exist
+                    if (tags && tags.length > 0 && tagsToAdd.length > 0) {
+                        await Promise.all(
+                            tagsToAdd.map(async (tagId) => {
+                                const sectionTag = new SectionTag({
+                                    sectionId: newSection.id,
+                                    tagId: tagId.id
+                                });
+
+                                await sectionTag.create(context);
+                            })
+                        );
+                    }
+
+                    // Return newly created section with tags
+                    return await Section.getSectionBySectionId('addSection resolver', context, sectionId);
+                }
+            }
+        },
+
+        updateSection: async (_, { input: { sectionId, name, introduction, requirements, guidance, tags, displayOrder } }, context: MyContext): Promise<Section> => {
+
+            // Get Section based on provided sectionId
+            const sectionData = await Section.getSectionBySectionId('section resolver', context, sectionId);
+
+            // Throw Not Found error if Section is not found
+            if (!sectionData) {
+                throw NotFoundError('Section not found')
+            }
+
+            // Check that user has permission to update this section
+            if (await hasPermission(context, sectionData.templateId)) {
+                const section = new Section({
+                    id: sectionData.id,
+                    templateId: sectionData.templateId,
+                    createdById: sectionData.createdById,
+                    name: name,
+                    introduction: introduction,
+                    requirements: requirements,
+                    guidance: guidance,
+                    displayOrder: displayOrder,
+                    isDirty: true  // Mark as dirty for update
+                });
+
+                const updatedSection = await section.update(context);
+
+                // If there are errors than throw a Bad User Input error
+                if (updatedSection.errors) {
+                    const errorMessages = updatedSection.errors.join(', ');
+                    throw BadUserInput(errorMessages);
+                } else {
+                    //Get all the existing tags associated with this section
+                    const existingTags = await Tag.getTagsBySectionId('updateSection resolver', context, sectionId);
```

Comment:
I think all of the logic to figure out what tags are new could be moved to the new `sectionService`.

You are accounting for new tags that might have been added, but are we also looking for tags that might have been removed?

Reply:
@briri  - That makes sense to move the check for which new tags are not duplicates of existing tags to the sectionService. As far as removing tags, I was assuming when a tag is removed, it is completely removed from the database table. If that is the case, I don't think we need to check for tags that were removed.

Reply:
Ok. If you we can open a new ticket if we find any issues.

Reply:
Thanks Brian. I actually initially made the mistake of grabbing the tags from the tags table, rather than the sectionTags table. What I was trying to do was to get all the records from the secionTags table with the sectionId, and then prevent duplicate entries. I fixed that in the last commit :) 

I'm sure we'll encounter issues and areas of improvement as we start using the resolvers for the new pages. I will be sure to add new tickets accordingly.

Thanks again for your review.